File: Sample/java/src/main/java/net/nend/sample/java/icon/IconAttachAndDetachActivity.java
package net.nend.sample.java.icon;

import android.os.Bundle;
import androidx.appcompat.app.AppCompatActivity;
import android.view.View;
import android.widget.LinearLayout;

import net.nend.android.NendAdIconLoader;
import net.nend.android.NendAdIconView;
import net.nend.sample.java.R;
import net.nend.sample.java.SampleConstants;

import java.util.Random;

public class IconAttachAndDetachActivity extends AppCompatActivity implements SampleConstants {

	private NendAdIconLoader mIconLoader;
	private LinearLayout mIconContainer;
	
	@Override
	protected void onCreate(Bundle savedInstanceState) {
		super.onCreate(savedInstanceState);
		setContentView(R.layout.icon_attach_detach);
		
		mIconContainer = (LinearLayout)findViewById(R.id.icon_container);
		
        NendAdIconView view1 = new NendAdIconView(getApplicationContext());
        NendAdIconView view2 = new NendAdIconView(getApplicationContext());
        NendAdIconView view3 = new NendAdIconView(getApplicationContext());
        NendAdIconView view4 = new NendAdIconView(getApplicationContext());
        
        mIconLoader = new NendAdIconLoader(getApplicationContext(), ICON_SPOT_ID, ICON_API_KEY);
        mIconLoader.addIconView(view1);
        mIconLoader.addIconView(view2);
        mIconLoader.addIconView(view3);
        mIconLoader.addIconView(view4);
        
        mIconContainer.addView(view1);
        mIconContainer.addView(view2);
        mIconContainer.addView(view3);
        mIconContainer.addView(view4);
        
        mIconLoader.loadAd();
	}
	
	public void onClickAdd(View v) {
		if ( 4 <= mIconLoader.getIconCount() ) {
			return;
		}
		NendAdIconView view = new NendAdIconView(getApplicationContext());
		mIconLoader.addIconView(view);
		mIconContainer.addView(view);
	}
	
	public void onClickRemove(View v) {
		if ( 0 == mIconLoader.getIconCount() ) {
			return;
		}
		Random random = new Random();
		int index = random.nextInt(mIconLoader.getIconCount());
		NendAdIconView view = (NendAdIconView)mIconContainer.getChildAt(index);
		mIconLoader.removeIconView(view);
		mIconContainer.removeViewAt(index);
	}
}

File: Sample/kotlin/src/main/java/net/nend/sample/kotlin/icon/IconAttachAndDetachActivity.kt
package net.nend.sample.kotlin.icon

import android.os.Bundle
import android.view.View
import androidx.appcompat.app.AppCompatActivity
import net.nend.android.NendAdIconLoader
import net.nend.android.NendAdIconView
import net.nend.sample.kotlin.databinding.IconAttachDetachBinding
import net.nend.sample.kotlin.icon.IconSampleActivity.Companion.ICON_API_KEY
import net.nend.sample.kotlin.icon.IconSampleActivity.Companion.ICON_SPOT_ID
import java.util.Random

class IconAttachAndDetachActivity : AppCompatActivity() {

    private lateinit var binding: IconAttachDetachBinding
    private lateinit var iconLoader: NendAdIconLoader

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = IconAttachDetachBinding.inflate(layoutInflater)
        setContentView(binding.root)

        val view1 = NendAdIconView(applicationContext)
        val view2 = NendAdIconView(applicationContext)
        val view3 = NendAdIconView(applicationContext)
        val view4 = NendAdIconView(applicationContext)

        iconLoader = NendAdIconLoader(applicationContext, ICON_SPOT_ID, ICON_API_KEY).apply {
            addIconView(view1)
            addIconView(view2)
            addIconView(view3)
            addIconView(view4)
            loadAd()
        }

        binding.iconContainer.run {
            addView(view1)
            addView(view2)
            addView(view3)
            addView(view4)
        }
    }

    @Suppress("UNUSED_PARAMETER")
    fun onClickAdd(view: View) {
        if (iconLoader.iconCount >= 4) {
            return
        }
        NendAdIconView(applicationContext).apply {
            iconLoader.addIconView(this)
            binding.iconContainer.addView(this)
        }
    }

    @Suppress("UNUSED_PARAMETER")
    fun onClickRemove(view: View) {
        if (iconLoader.iconCount == 0) {
            return
        }
        val index = Random().nextInt(iconLoader.iconCount)
        val iconView = binding.iconContainer.getChildAt(index) as NendAdIconView
        iconLoader.removeIconView(iconView)
        binding.iconContainer.removeViewAt(index)
    }
}